### ISSUE TITLE
Fine-grained permission control per team

### ISSUE THREAD

**Issue by k2nr** [degica/barcelona#134]
The current permission control is so simple that we cannot control user's accessibility to districts and service. Per-user permission is I think overkill so I propose per-team permission as follows.
## `Team` resource

I propose to add `Team` resource(model) which holds the following permissions:

| Name | Default | Description |
| --- | --- | --- |
| districts | `[]` | a list of districts which team members can access to |
| role | `member` | One of `admin`, `developer`, or `readonly` |
| github_team |  | A name of GitHub's team which Barcelona team is linked to |
| github_organization |  | A name of GitHub's organization which Barcelona team is linked to |
## `User`
- Each user can belong to multiple teams
- User should first provide GitHub token
  - Barcelona get github information about the user and decides which teams the user will belong to
## Roles

Each `Team` has one of the following roles.
### `admin` role

`admin` role can do anything to districts the team belong to
### `developer` role

`developer` role can do anything `admin` can do **except** following actions:
- updating / deleting / creating a district
- launch / terminate a container instance
- deleting a heritage
- updating / creating / deleting a team
- updating / deleting a user (except himself)
### `readonly` role

Maybe we don't need this role?

`readonly` role can do only the following actions:
- show a district
- show a heritage
- show a team
- show a user

@essa what do you think about this? Is this too complicated or not so flexible?

**Comment by k2nr** [comment]
Closing in favor of #401
